wyvern/feature_store/feature_server.py

```diff
@@ -456,6 +462,62 @@ async def get_historical_features(
             results=final_df.to_dict(orient="records"),
         )
 
+    @app.post(f"{settings.WYVERN_HISTORICAL_FEATURES_PATH}_v2")
+    async def get_historical_features_v2(
+        data: GetHistoricalFeaturesRequestV2,
+    ) -> GetHistoricalFeaturesResponseV2:
+        # Generate a 10-digit hex for the request
+        random_id = secrets.token_hex(5)
+
+        # convert the data input to pandas dataframe
+        realtime_features, feast_features = separate_real_time_features(data.features)
+        valid_realtime_features: List[str] = []
+        composite_entities: Dict[str, List[str]] = {}
+        for realtime_feature in realtime_features:
+            entity_type_column = RealtimeFeatureComponent.get_entity_type_column(
+                realtime_feature,
+            )
+            entity_names = RealtimeFeatureComponent.get_entity_names(realtime_feature)
+            if not entity_type_column or not entity_names:
+                logger.warning(f"feature={realtime_feature} is not found")
+                continue
+
+            if len(entity_names) == 2:
+                composite_entities[entity_type_column] = entity_names
+            valid_realtime_features.append(realtime_feature)
+
+        composite_columns = ",".join(
+            [
+                " || ':' || ".join(entities) + f" AS {entity_type_column}"
+                for entity_type_column, entities in composite_entities.items()
+            ],
+        )
+        composite_historical_feature_table = f"HISTORICAL_FEATURES_{random_id}"
+
+        select_sql = f"""
+        CREATE TEMPORARY TABLE {composite_historical_feature_table} AS
+        SELECT *, {composite_columns}, TIMESTAMP as event_timestamp
+        FROM {data.table}
+        """
+        snowflake_ctx = generate_snowflake_ctx()
+        snowflake_ctx.cursor().execute(select_sql)
```

Comment:
should we handle some errors and add retry logic?

Reply:
I'm not sure when we should retry tho. Let's start with busting the API if there's any error. most likely error comes from bad data in the input table (entity table), which probably wont' be fixed by retry. 

let's launch this version without error handling and start catching issues with matias' data.